src/ansys/aedt/core/modules/mesh.py

```diff
@@ -1101,35 +1101,38 @@ def assign_skin_depth(
         else:
             name = generate_unique_name("SkinDepth")
 
-        if maximum_elements is None:
-            restrictlength = False
-            maximum_elements = "1000"
+        if self._app.design_type == "Maxwell 2D":
+            props = OrderedDict(
+                {
+                    "Edges": assignment,
+                    "SkinDepth": skin_depth,
+                    "NumLayers": layers_number,
+                }
+            )
         else:
-            restrictlength = True
-        assignment = self._app.modeler.convert_to_selections(assignment, True)
+            if maximum_elements is None:
+                restrictlength = False
+                maximum_elements = "1000"
+            else:
+                restrictlength = True
 
-        if isinstance(assignment[0], int):
-            seltype = "Faces"
-        elif isinstance(assignment[0], str):
-            seltype = "Objects"
-        else:
-            seltype = None
-        if seltype is None:
-            self.logger.error("Error in Assignment")
-            return
+            if isinstance(assignment[0], int):
+                seltype = "Faces"
+            elif isinstance(assignment[0], str):
+                seltype = "Objects"
```

Comment:
Following the modified lines of code, shouldn't we log an error here is the assigment[0] is not int or str ?
Otherwise, the error that the user will see might be confusing (even if we asked for a list of string or int in the method description).

Reply:
@SMoraisAnsys I removed it because assignment will always be either an int or a string. If you look previous lines, assignment is computed here:
`assignment = self.modeler.convert_to_selections(assignment, True)`
which returns always an int or a string. It's not the assignment passed by the user.
Hope this makes sense :)